### PR TITLE
Write functions to run default task on OSG

### PR DESCRIPTION
**Pull request by mingtaiha**

### CODE REVIEW

File: src/aimes/emgr/emgr.py [hunk 1]
```diff
@@ -309,6 +309,41 @@ def derive_pilot_descriptions(cfg, strategy):
...
+    pdescs = list()
+
+    #   Creating the specified number of OSG pilots. Each pilot is given
+    #   the project ID, resource, number of cores, and wall time
+    for pilot in range(cfg['num_osg_pilots']):
+
+        pdesc = rp.ComputePilotDescription()
+
+        pdesc.project = cfg['project_ids'].get('osg.xsede-virt-clust')
+        pdesc.resource = 'osg.xsede-virt-clust'
```
Comment (mturilli) [inline src/aimes/emgr/emgr.py]: I would probably not hard-code the name of the resource in this function. Maybe better using a configuration variable? Also, maybe using what is already in place, i.e., unsupported resource so to have a link with the (future) interaction with bundles?

File: src/aimes/emgr/emgr.py [hunk 2]
```diff
@@ -309,6 +309,41 @@ def derive_pilot_descriptions(cfg, strategy):
...
+
+        print "DEBUG: strategy['inference'] = %s" % strategy['inference']
+
+        pdesc.cores = 1
```
Comment (mturilli) [inline src/aimes/emgr/emgr.py]: I would probably move also this to a configuration parameter. The rationale: we may want to test what happens when we ask for 2 cores per pilot on OSG--e.g., super-long Tq? No resources available?

File: src/aimes/emgr/strategy.py [hunk 3]
```diff
@@ -114,7 +115,74 @@ def derive_execution_stategy_skeleton(cfg, workload, resources, run):
...
+
+    #   Defining the percentage of available resources to use
+    strategy['heuristic']['percentage_resources'] = cfg['pct_resources']
+
+    #   Initializing list of resources to use
+    strategy['inference']['target_resources'] = list()
+
+    #   Defining resources to use for resources which do and do not support bundles
+    for i in range(cfg['num_osg_pilots']):
+        strategy['inference']['target_resources'].append('osg.xsede-virt-clust')
```
Comment (mturilli) [inline src/aimes/emgr/strategy.py]: See my note on hard-coding OSG resource name in src/aimes/emgr/emgr.py

File: src/aimes/emgr/strategy.py [hunk 4]
```diff
@@ -114,7 +115,74 @@ def derive_execution_stategy_skeleton(cfg, workload, resources, run):
...
+        strategy['inference']['number_pilots'] = max(int(cfg['num_osg_pilots']), 1)
+        assert strategy['inference']['number_pilots'] >= 1, "Must specify a positive integer for the number of OSG pilots to submit\n"
+            
+    #   Choosing the scheduler for the CUs. If there is only one pilot. then
+    #   submit directly to the pilot. If there is more than one pilot. The number
+    #   of pilots has been asserted to be a strictly positive integer
+    if strategy['inference']['number_pilots'] > 1:
+        strategy['inference']['rp_scheduler'] = 'SCHED_BACKFILLING'
+    else:
+        strategy['inference']['rp_scheduler'] = 'SCHED_DIRECT_SUBMISSION'
```
Comment (mturilli) [inline src/aimes/emgr/strategy.py]: We may have to check with Andre whether SCHED_DIRECT_SUBMISSION is still valid. I think they are disbanding direct submission in favor of round robin.

File: src/aimes/emgr/strategy.py [hunk 5]
```diff
@@ -114,7 +115,74 @@ def derive_execution_stategy_skeleton(cfg, workload, resources, run):
...
+
+    #   On OSG, it is not necessary to submit a wall time. However, pilots on RP
+    #   require a walltime, so 6 hours is given
+    strategy['inference']['compute_time_workload'] =  21600 # 6 Hours in seconds
```
Comment (mturilli) [inline src/aimes/emgr/strategy.py]: We should probably specify this in a configuration parameter.

File: src/aimes/emgr/strategy.py (hunk 4 above)
Comment (mingtaiha) [inline src/aimes/emgr/strategy.py]: I am fine with either Direct Submission or Round Robin. If we choose to use Round Robin instead, we will need to make the corresponding changes in XSEDE-equivalent of the function

**Comment by mingtaiha** [comment]
I can see where you are coming from with all the comments which try to avoid hard-coding values, or simply exposing a parameter to the user (us, namely). We can add a separate entry in the experiment JSON file as the least invasive way of exposing this functionality. This we, we can set the values of parameters which are either hard-coded to the user.

File: src/aimes/emgr/emgr.py (hunk 1 above)
Comment (mingtaiha) [inline src/aimes/emgr/emgr.py]: By using what is already in place, do you mean for me to parse whether the resource is in the resource (OSG) either supports or not supports (unsupported) OSG?

File: src/aimes/emgr/strategy.py (hunk 4 above)
Comment (andre-merzky) [inline src/aimes/emgr/strategy.py]: DIRECT_SUBMISSION is not available in the split_module branch -- and would make only limited sense on OSG anyway, as it only 'schedules' over one pilot.  The new RR scheduler is semantically equivalent to the former direct scheduler in the case of one pilot.

Comment (mturilli) [inline src/aimes/emgr/strategy.py]: Thank you Andre, agreed. Then let's transition to RR.

File: src/aimes/emgr/emgr.py (hunk 1 above)
Comment (mturilli) [inline src/aimes/emgr/emgr.py]: I mean something like this:

```
    "bundle": {
        "mongodb"  : "mongodb://144.76.72.175:24242/swift_aimes_bundle",
        "template" : "./conf/bundle.template",
        "origin"   : "144.76.72.175",
        "username" : "mingtaiha",
        "resources": {
            # "supported": {
            #     "hopper.nersc.gov"          : "pbs",
            #     "stampede.tacc.xsede.org"   : "slurm",
            #     "comet.sdsc.xsede.org"      : "slurm",
            #     "supermic.cct-lsu.xsede.org": "pbs",
            #     "trestles.sdsc.xsede.org"   : "pbs",
            #     "blacklight.psc.xsede.org"  : "pbs",
            #     "gordon.sdsc.xsede.org"     : "pbs"
            # },
            # "unsupported": {
            #   "supermic.cct-lsu.xsede.org": {
            #     "sched": "pbs",
            #     "fconf": "conf/xsede.supermic.json"
            #   },
            #     "comet.sdsc.xsede.org": {
            #       "sched": "slurm",
            #       "fconf": "conf/xsede.comet.json"
            #     },
            #     "gordon.sdsc.xsede.org": {
            #       "sched": "pbs",
            #       "fconf": "conf/xsede.gordon.json"
            #      },
                 "xd-login.opensciencegrid.org": {
                   "sched": "condor",
                   "fconf": "conf/osg.xsede-virt-clust.json"
                  },
                }
             }
           }
```

Comment (mingtaiha) [inline src/aimes/emgr/emgr.py]: Oh, I already do that in the JSON file. That should not be a problem.

**Pull request merged**